Server/ConcordiaCurriculumManager/Repositories/CourseRepository.cs

```diff
@@ -65,5 +68,6 @@ public async Task<bool> SaveCourse(Course course)
         .Include(course => course.SupportingFiles)
         .Include(course => course.CourseCourseComponents)
         .OrderByDescending(course => course.Version)
+        .Select(ObjectSelectors.CourseSelector())

```

Comment:
Rather than making selectors on all the repository methods, it might make more sense to just .Include CourseReferenced to the ones that need it. With the course selector, we're re-downloading all the supporting files everywhere.

Reply:
If I do that, the course references won't include the course that is referencing the current course (it would just include the Id of that course). If I remove the supporting files from the selector, would that suffice?